static/js/components/forms/B2BPurchaseForm.js

```diff
@@ -49,6 +49,13 @@ export const validate = (values: Object) => {
 }
 
 class B2BPurchaseForm extends React.Component<Props> {
+  componentDidMount() {
+    const applyCouponButton = document.getElementById("apply-coupon")
```

Comment:
It is not recommended in React to get DOM element in this way please fetch via reference from the node trees.

Reply:
sorry?
This way of fetching the node is already being used in the project at other places.

Reply:
Please Make use of refs to access the DOM element. It is not good practice to approach `getElementById`

Reply:
ok, let me try that if that works out.